bundles/org.openhab.binding.evcc/src/main/java/org/openhab/binding/evcc/internal/EvccHandler.java

```diff
@@ -398,123 +533,147 @@ private void updateChannelsLoadpoint(int loadpointId) {
 
         boolean vehiclePresent = loadpoint.getVehiclePresent();
-        channel = new ChannelUID(uid, loadpointName, CHANNEL_LOADPOINT_VEHICLE_PRESENT);
+        channel = new ChannelUID(uid, channelGroup, CHANNEL_LOADPOINT_VEHICLE_PRESENT);
         updateState(channel, OnOffType.from(vehiclePresent));
 
         float vehicleRange = loadpoint.getVehicleRange();
-        channel = new ChannelUID(uid, loadpointName, CHANNEL_LOADPOINT_VEHICLE_RANGE);
+        channel = new ChannelUID(uid, channelGroup, CHANNEL_LOADPOINT_VEHICLE_RANGE);
         updateState(channel, new QuantityType<>(vehicleRange, MetricPrefix.KILO(SIUnits.METRE)));
 
         float vehicleSoC = loadpoint.getVehicleSoC();
-        channel = new ChannelUID(uid, loadpointName, CHANNEL_LOADPOINT_VEHICLE_SOC);
+        channel = new ChannelUID(uid, channelGroup, CHANNEL_LOADPOINT_VEHICLE_SOC);
         updateState(channel, new QuantityType<>(vehicleSoC, Units.PERCENT));
 
-        String vehicleTitle = loadpoint.getVehicleTitle();
-        channel = new ChannelUID(uid, loadpointName, CHANNEL_LOADPOINT_VEHICLE_TITLE);
-        updateState(channel, new StringType(vehicleTitle));
+        String vehicleName = loadpoint.getVehicleName();
+        channel = new ChannelUID(uid, channelGroup, CHANNEL_LOADPOINT_VEHICLE_NAME);
+        updateState(channel, new StringType(vehicleName));
+    }
+
+    private void updateChannelsVehicle(String vehicleName) {
+        final Result result = this.result;
+        if (result == null) {
+            return;
+        }
+        final ThingUID uid = getThing().getUID();
+        final String channelGroup = "vehicle" + vehicleName;
+        ChannelUID channel;
+        Vehicle vehicle = result.getVehicles().get(vehicleName);
+
+        String title = vehicle.getTitle();
+        channel = new ChannelUID(uid, channelGroup, CHANNEL_VEHICLE_TITLE);
+        updateState(channel, new StringType(title));
+
+        float minSoC = vehicle.getMinSoC();
+        channel = new ChannelUID(uid, channelGroup, CHANNEL_VEHICLE_MIN_SOC);
+        updateState(channel, new QuantityType<>(minSoC, Units.PERCENT));
+
+        float limitSoC = vehicle.getLimitSoC();
+        channel = new ChannelUID(uid, channelGroup, CHANNEL_VEHICLE_LIMIT_SOC);
+        updateState(channel, new QuantityType<>(limitSoC, Units.PERCENT));
+
+        Plan plan = vehicle.getPlan();
+        if (plan == null && vehiclePlans.get(vehicleName) == null) {
+            vehiclePlans.put(vehicleName, new Triple<>(false, 100f, ZonedDateTime.now().plusHours(12)));
+        } else if (plan != null) {
+            vehiclePlans.put(vehicleName, new Triple<>(true, plan.getSoC(), ZonedDateTime.parse(plan.getTime())));
+        }
+        updateVehiclePlanChannel(vehicleName, uid, channelGroup, channel);
+    }
+
+    private void updateVehiclePlanChannel(String vehicleName, ThingUID uid, String channelGroup, ChannelUID channel) {
```

Comment:
This argument is never used, and the passed in value (see comment above) is wrong anyway.

```suggestion
    private void updateVehiclePlanChannel(String vehicleName, ThingUID uid, String channelGroup) {
```

Reply:
ack

Reply:
Seems like the changes for these comments are missing ...

Reply:
https://github.com/openhab/openhab-addons/pull/16114/commits/c9da7d6b9ea827957da6ac83b0d408d3d5f88d05

Reply:
It is not in the commit.

Reply:
I guess these are the 3 changes that we are talking about?

![image](https://github.com/openhab/openhab-addons/assets/44850211/7665671a-b359-427f-ad8d-d71413085a6e)

Reply:
you are right . the issue is on my end. I will fix ...

Reply:
Now it's committed - sorry
https://github.com/openhab/openhab-addons/pull/16114/commits/2a0ac93e119a2331d2998d97f9c5703217bcc8d0